dependencies.py

```diff
@@ -85,6 +84,7 @@ def extra_requirements() -> dict[str, list[str]]:
         {
             "dev-postgres": ["pytest-mock-resources[postgres]~=2.12"],
             "kfp18": ["mlrun_pipelines_kfp_v1_8[kfp]~=0.6.0"],
+            "mlflow": ["mlflow~=3.0.0"],
```

Comment:
If we went all the way up to 3, why ~=3.0.0 and not 3.8.0? Does something break for us between 3.0 and 3.8?

Reply:
in the ticket it said to move to version 3.0.0. Anyway this change should support later versions as well.